Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -675,6 +676,23 @@ extension ByteBuffer {
     }
 }
 
+extension ByteBuffer: Codable {
+
+    /// Creates a ByteByffer by decoding from a Base64 encoded single value container.
+    public init(from decoder: Decoder) throws {
+        let container = try! decoder.singleValueContainer()
+        let base64String = try! container.decode(String.self)
+        self = try! ByteBuffer(bytes: base64String.base64Decoded())
+    }
+
+    /// Encodes this buffer as a base64 string in a single value container.
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        let base64String = String(base64Encoding: self.readableBytesView)
+        try! container.encode(base64String)
```

Comment:
this `try!` should also be a `try`. I don't think JSONEncoder will ever throw on a string but there might be other encoders that limit the size of the output of a single string value (I think).

Reply:
Also done: https://github.com/apple/swift-nio/pull/2509/commits/3861cf8b76686fa907d44fb96bf6565483bc5185 

Yep, that was prototype-grade spike at first.